module-services/service-cellular/DTMFCode.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "DTMFCode.hpp"
+#include <Utils.hpp>
+#include <stdexcept>
+
+DTMFCode::DTMFCode(const char digitChar)
+{
+    if ((digitChar >= '0' && digitChar <= '9') || digitChar == '*' || digitChar == '#') {
+        this->digitChar = digitChar;
+    }
+    else {
+        this->digitChar = 0;
+        throw std::out_of_range("Invalid number!");
```

Comment:
please add some more information (i.e "DTMF: Invalid number" in string just in case (as we are using basic exception)

Reply:
sure, but wouldn't the trace show the file either way?

Reply:
added